packages/core/data-transfer/src/file/providers/source/index.ts

```diff
@@ -81,18 +111,28 @@ class LocalFileSourceProvider implements ISourceProvider {
       }
       throw new ProviderInitializationError(`File '${filePath}' is not a valid Strapi data file.`);
     }
+
+    if (!this.#metadata) {
+      throw new ProviderInitializationError('Could not load metadata from Strapi data file.');
+    }
   }
 
-  getMetadata() {
-    // TODO: need to read the file & extract the metadata json file
-    // => we might also need to read the schema.jsonl files & implements a custom stream-check
+  async getMetadata() {
+    if (this.#metadata) {
+      return this.#metadata;
```

Comment:
Could you describe what edge-case we're handling there?

Reply:
After reading the code, I assume it's made to cache the jsonl read operation?

Reply:
Yeah, we're reading metadata three times so this way it's caching it. I'll restructure it into different loading/retreiving methods)

Reply:
Alright make sense, might be worth adding a small comment to explain the benefit :slightly_smiling_face: